controllers/vm_controller.go

```diff
@@ -0,0 +1,140 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/hashicorp/go-multierror"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	kubevirtv1 "kubevirt.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"kubevirt.io/ssp-operator/pkg/monitoring/metrics"
+)
+
+// +kubebuilder:rbac:groups=kubevirt.io,resources=virtualmachines,verbs=get;list;watch
+// +kubebuilder:rbac:groups=core,resources=persistentvolumeclaims,verbs=get;list;watch
+// +kubebuilder:rbac:groups=core,resources=persistentvolumes,verbs=get;list;watch
+
+const vmControllerName = "vm-controller"
+
+type VmReconciler struct {
+	client client.Client
+	log    logr.Logger
+}
+
+func CreateVmController(mgr ctrl.Manager) (*VmReconciler, error) {
+	return newVmReconciler(mgr)
+}
+
+func (r *VmReconciler) Name() string {
+	return vmControllerName
+}
+
+func (r *VmReconciler) Start(_ context.Context, mgr ctrl.Manager) error {
+	return r.setupController(mgr)
+}
+
+func newVmReconciler(mgr ctrl.Manager) (*VmReconciler, error) {
+	logger := ctrl.Log.WithName("controllers").WithName("VirtualMachines")
+	reconciler := &VmReconciler{
+		client: mgr.GetClient(),
+		log:    logger,
+	}
+
+	return reconciler, nil
+}
+
+func (r *VmReconciler) setupController(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(vmControllerName).
+		For(&kubevirtv1.VirtualMachine{}).
+		Complete(r)
+}
+
+func (r *VmReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.Result, err error) {
+	r.log.Info("Starting vm reconciliation...", "request", req.String())
```

Comment:
This log line will be very verbose, because it will be printed for every change of every VM in the cluster.
Please remove it.

Reply:
removed